### PR TITLE
When new webservice key is created, add log

### PR DESCRIPTION
**Pull request by ksaandev**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | I add a call to PrestaShopLogger in webservice key creation so it's logged. I think for security it's important BO Admin can know who create keys. Keys can provide access to critical data.
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Create a new Webservice Key then go in Logs page and see latest log which says "Webservice key created by employee Ksaan Ami (ksaan.ami@gmail.com)"

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19831)

### CODE REVIEW

**Review by NeOMakinG — DISMISSED** [review]
Good thing I think, approving for the feature :)

**Review by matks — DISMISSED** [review]
Indeed it's an important event to be logged

However ... we should then start logging also webservice keys updates and deletes, right ? 😄 

Let's no make the scope of this PR bigger, but this is something we could add in the future 😊

**Review by Progi1984 — DISMISSED** [review]

**Review by Junebyun — DISMISSED** [review]

**Comment by Junebyun** [comment]
Hey guys, just to check: will the text _"Webservice key created by employee..."_ be shown like a **notification** (shortly shown on a banner) or like a **text block** in the page (text which will be permanently there if you go to the Logs page)? 

Could you show me a screenshot of how it will be shown?

Because if it turns out to be a permanent text in the Logs page I'd rather put the domain as _Feature_ instead of _Notifications_.

**Comment by ksaandev** [comment]
> Let's no make the scope of this PR bigger, but this is something we could add in the future 😊

I can do it in another PR :)

**Comment by ksaandev** [comment]
> Hey guys, just to check: will the text _"Webservice key created by employee..."_ be shown like a **notification** (shortly shown on a banner) or like a **text block** in the page (text which will be permanently there if you go to the Logs page)?

No, it's different. It will appear as one item of the logs list.

The Logs page show multiple logs which are like registry items. You can add as many as you want, it's things you want to register and be able to search, sort, list later when you want to understand what happened in the past.

So it will not be a banner, it will not be a block of text. It will be one item that appears inside the list displayed in the Logs page.

File: classes/webservice/WebserviceKey.php [hunk 1]
```diff
@@ -53,7 +53,23 @@ public function add($autodate = true, $nullValues = false)
...
+        if ($result) {
+            PrestaShopLogger::addLog(
+                Context::getContext()->getTranslator()->trans(
+                    'Webservice key created by employee %s %s (%s)',
+                    [
+                        Context::getContext()->employee->firstname,
+                        Context::getContext()->employee->lastname,
+                        Context::getContext()->employee->email,
+                    ],
+                    'Admin.Advparameters.Notification'
```
Comment (Junebyun) [inline classes/webservice/WebserviceKey.php]:
```suggestion
                    'Webservice key created by employee %s %s (%s)',
                    [
                        Context::getContext()->employee->firstname,
                        Context::getContext()->employee->lastname,
                        Context::getContext()->employee->email,
                    ],
                    'Admin.Advparameters.Feature'
```

**Comment by Junebyun** [comment]
> > Hey guys, just to check: will the text _"Webservice key created by employee..."_ be shown like a **notification** (shortly shown on a banner) or like a **text block** in the page (text which will be permanently there if you go to the Logs page)?
> 
> No, it's different. It will appear as one item of the logs list.
> 
> The Logs page show multiple logs which are like registry items. You can add as many as you want, it's things you want to register and be able to search, sort, list later when you want to understand what happened in the past.
> 
> So it will not be a banner, it will not be a block of text. It will be one item that appears inside the list displayed in the Logs page.

Thank you for your explanation, it's very clear to me now! :)
I would suggest _Admin.Advparameters.Feature_ for its domain then.

**Review by matks — DISMISSED** [review]

**Review by Junebyun — DISMISSED** [review]

File: classes/webservice/WebserviceKey.php [hunk 2]
```diff
@@ -53,7 +53,23 @@ public function add($autodate = true, $nullValues = false)
...
+
+        if ($result) {
+            PrestaShopLogger::addLog(
+                Context::getContext()->getTranslator()->trans(
+                    'Webservice key created by employee %s %s (%s)',
+                    [
+                        Context::getContext()->employee->firstname,
+                        Context::getContext()->employee->lastname,
+                        Context::getContext()->employee->email,
+                    ],
```
Comment (PierreRambaud) [inline classes/webservice/WebserviceKey.php]:
```suggestion
                Context::getContext()->getTranslator()->trans(
                    'Webservice key created by employee "%d" %s %s (%s)',
                    [
                        Context::getContext()->employee->id,
                        Context::getContext()->employee->firstname,
                        Context::getContext()->employee->lastname,
                        Context::getContext()->employee->email,
                    ],
```

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Should be better to also add the id, employee are able to change their firstname, lastname and email before running the key creation.

File: classes/webservice/WebserviceKey.php (hunk 2 above)
Comment (Progi1984) [reply classes/webservice/WebserviceKey.php]: I approve.

Comment (ksaandev) [reply classes/webservice/WebserviceKey.php]: Thank you ! done

**Review by matks — APPROVED** [review]

**Review by LouiseBonnard — APPROVED** [review]

**Review by Junebyun — APPROVED** [review]

**Comment by SD1982** [comment]
HI @ksaandev some changes have been requested by @PierreRambaud can you check this please ?

**Review by PierreRambaud — APPROVED** [review]

**Comment by SD1982** [comment]
Oups sorry @ksaandev  my bad it's good for the requested changes !

**Comment by SD1982** [comment]
LGTM the log is correctly added ! Thanks @ksaandev !!

**Comment by PierreRambaud** [comment]
Thanks @ksaandev

**Comment by ksaandev** [comment]
> LGTM the log is correctly added ! Thanks @ksaandev !!

😄 thank you guys it's so cool to see PRs merged very fast

**Pull request merged**